src/bond/james.cljc

```diff
@@ -47,3 +63,25 @@
                                   [v `(spy (constantly nil))])) vs)
                       (vec))
      ~@body))
+
+(defmacro with-stub-namespaces
+  "Takes a vector of namespaces and/or [namespace replacement] vectors.
+
+  Replaces every fn in each namespace with its replacement. If a replacement is
+  not specified for a namespace every fn in that namespace is replaced with
+  (constantly nil). All replaced functions are also spied."
+  [namespaces & body]
+  `(with-redefs ~(->> namespaces
```

Comment:
Bikeshed:

I would personally split the macros into a single-line macro and a function, like `clojure.core`'s `future` and `future-call`.